xarray/core/variable.py

```diff
@@ -333,7 +333,7 @@ def shape(self):
 
     @property
     def nbytes(self):
-        return self.size * self.dtype.itemsize
+        return self.data.nbytes
```

Comment:
I think we'll want to account for `nbytes` not existing on `self.data` (it's not in the [array API](https://data-apis.org/array-api/latest/API_specification/array_object.html#attributes)) So something like

```
if hasattr(self.data, "nbytes"):
    ...
else:
    return self.size * self.dtype.itemsize
```